
![alt text](assets/Capture.PNG?raw=true "header")


## **Introduction**

UVRI and MRC-University of Glasgow Center for Virus Research have organised a one day joint bioinformatics training happening on 26th Feb 2019. The training will focus on genome assemblies and taxonomic profiling from NGS data. A quick intro to linux will be offered as well.
The training is aimed at graduate students and other researchers. You don't need to have any previous knowledge of the tools that will be presented at the training.
UVRI Library, Plot 51-59, Nakiwogo Road Entebbe on Feb 26, 2019. Participants owning laptops can bring them along.
They do not need to have specific software packages installed since we sahll be using the UVRI server.

Contact: Please email <a href="jkayondo@uvri.go.ug">jkayondo@uvri.go.ug<a/> or <a href="assekagiri@uvri.go.ug">assekagiri@uvri.go.ug<a/> for more information.

Collaborative Notes will be posted here using <a href="http://pad.software-carpentry.org/2019-02-19-uvri">Etherpad</a>


## **Outline**

* Linux basics
* Tools required
* Tools for analysing metagenomic data
* Metagenome assembly
* Taxonomic identification
* Visualisation

## **Sessions**

| Time  | Title|
| ------------- | ------------- |
|0900:0915 |     Welcome and participants Introduction|
|0915:1015 |     Introduction to Linux |
|1015:1045 |      NGS file formats and Quality control (Theory/Practical)
|1045:1115 |      Coffee break|
|1115:1215 |      Reference mapping (Theory/Practical)|
|1215: 1245 |     Assembly statistics, visualisation and consensus generation (Theory)|
|1245:1330 |      Lunch|
|1330:1400 |      Assembly statistics, visualisation and consensus generation (Practical)|
|1400:1430 |      DeNovo assembly (Theory)|
|1430:1500 |      DeNovo assembly (Practical)|
|1530:1600 |      Coffee break|
|1600:1630 |      Contig consolidation and Gap fillings|
|1630:1700 |      Kraken classification of NGS reads |





